src/pooler.erl

```diff
@@ -220,18 +229,25 @@ take_group_member(GroupName) ->
             {_, _, X} = os:timestamp(),
             Idx = (X rem length(Pools)) + 1,
             {PoolPid, Rest} = extract_nth(Idx, Pools),
-            take_first_pool([PoolPid | Rest])
+            case {take_first_pool([PoolPid | Rest], 0), Timeout} of
```

Comment:
The 1000 pools was just a crazy max case kind of thing, in practice we run one pool per Riak DB (so typically 5) and configure them with a init capacity of 10 and a max of 100.  When I first ran the change without the first pass of 0 through our load simulator I was seeing pools sizes roughly round 25, 20, 15, 10, 10 and after the change it evened out (around 14 per pool give or take one).  In the end nothing that was so critical that the system became unstable but I did find that typically a few servers received more load than others.